### PR TITLE
Dev/henry/10036 memberships update

### PR DESCRIPTION
**Pull request by henry739**
Documentation updates, shouldn't really be merged until we're releasing the new UI.

Separately, I'm working on updates to the OpenAPI annotations in Proto definitions to properly describe the endpoints and message types.

Updates to the Python SDK, YAML runner and Postman collections will also be made to avoid loss of functionality.

### CODE REVIEW

File: content/developers/api-reference/members-api/index.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+```bash
+curl -v -X GET \
+     -H "@$HOME/.datatrails/bearer-token.txt" \
+     https://app.datatrails.ai/archivist/iam/v1/members?member_state=MEMBER_STATE_BOTH
+```
+
+### Promoting a member of your tenant
+
+You can manage the roles of a user (i.e. promoting or demoting them by way of the 'OWNER' role) by
```
Comment (honourfish) [inline content/developers/api-reference/members-api/index.md:49]:
```suggestion
You can manage the roles of a member (i.e. promoting or demoting them by way of the 'OWNER' role) by
```

File: content/developers/api-reference/members-api/index.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+curl -v -X PATCH \
+    -H "@$HOME/.datatrails/bearer-token.txt" \
+    -H "Content-type: application/json" \
+    -d "{'roles': ['OWNER']}" \
+    https://app.datatrails.ai/archivist/iam/v1/{member_identity}
+```
+
+### Deactivating a member of your tenant
+
+When a user should no longer have access to your tenant, you can deactivate them progamatically
```
Comment (honourfish) [inline content/developers/api-reference/members-api/index.md:64]:
```suggestion
When a member should no longer have access to your tenant, you can deactivate them progamatically 
```

Comment (honourfish) [inline content/developers/api-reference/members-api/index.md:64]:
```suggestion
When a user should no longer have access to your tenant, you can deactivate them programatically 
```

File: content/developers/api-reference/members-api/index.md (hunk 1 above)
Comment (honourfish) [reply content/developers/api-reference/members-api/index.md:49]: its a bit jarring how the UI refers to these as users, but the API as members? maybe we want to be consistent at least on a customer facing level? if so we can disregard this suggestion?

Comment (henry739) [reply content/developers/api-reference/members-api/index.md:49]: What are your thoughts on this one @JAG-UK ? I know using the term 'Users' in the UI was requested. Is it worth the inconsistency overall?

Comment (JAG-UK) [reply content/developers/api-reference/members-api/index.md:49]: They're conceptually different: a user having roles is about what a person (or piece of software) can do within the system. The way we express and manage that is through Tenancy Membership. So a 'user' is a 'member' of a number of 'tenancies'.

I'll make a suggestion in a separate comment

File: content/developers/api-reference/members-api/index.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+By default this doesn't show members that are deactivated. You can filter the list to include 
+deactivated users:
+
+```bash
+curl -v -X GET \
+     -H "@$HOME/.datatrails/bearer-token.txt" \
+     https://app.datatrails.ai/archivist/iam/v1/members?member_state=MEMBER_STATE_BOTH
+```
+
+### Promoting a member of your tenant
```
Comment (JAG-UK) [inline content/developers/api-reference/members-api/index.md:47]:
```suggestion
### Promoting a user to 'OWNER' role in a tenancy
```

File: content/developers/api-reference/members-api/index.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+You can manage the roles of a user (i.e. promoting or demoting them by way of the 'OWNER' role) by
+PATCHing their record. With the identity of a member in your tenant to hand, submit this request
+to set their roles to a list containing only 'OWNER'. To demote the member, simply send an empty
+list.
```
Comment (JAG-UK) [inline content/developers/api-reference/members-api/index.md:52]:
```suggestion
You can manage the roles of a user (i.e. promoting or demoting them with the 'OWNER' role) by
PATCHing their membership record. 
With the identity of the tenant member record corresponding to the user in question, submit this request
to set their roles to a list containing only 'OWNER'. To demote the member, simply send an empty
list.
```

**Review by JAG-UK — CHANGES_REQUESTED** [review]

File: content/developers/api-reference/members-api/index.md (hunk 1 above)
Comment (henry739) [reply content/developers/api-reference/members-api/index.md:49]: Users only have roles in the context of a particular tenant though, i.e. their membership

**Review by JAG-UK — APPROVED** [review]

**Review by SteveLasker — APPROVED** [review]
LGTM
I focused on the integrations section, as I didn't run the new UX.

**Pull request merged**